tuf/data/targets.go

```diff
@@ -107,7 +107,10 @@ func (t *SignedTargets) BuildDelegationRole(roleName string) (DelegationRole, er
 				pubKey, ok := t.Signed.Delegations.Keys[keyID]
 				if !ok {
 					// Couldn't retrieve all keys, so stop walking and return invalid role
-					return DelegationRole{}, ErrInvalidRole{Role: roleName, Reason: "delegation does not exist with all specified keys"}
+					return DelegationRole{}, ErrInvalidRole{
+						Role:   roleName,
+						Reason: "delegation does not exist with all specified keys, " + keyID + " missing",
```

Comment:
Thank you so much for adding more information!  I was wondering if we could clarify this by saying something like `"role lists unknown key " + keyID + " as a signing key"` or something?